cmd/root.go

```diff
@@ -70,6 +74,46 @@ func getDefaultDeployment() types.RpmOstreeDeployment {
 	return rosState.Deployments[0]
 }
 
+// getRefDigest parses a Docker/OCI image reference and returns
+// its digest, or an error if the string fails to parse as
+// a "canonical" image reference with a digest.
+func getRefDigest(ref string) (string, error) {
+	refParsed, err := imgref.ParseNamed(ref)
+	if err != nil {
+		return "", fmt.Errorf("parsing reference: %q: %v", ref, err)
+	}
+	canon, ok := refParsed.(imgref.Canonical)
+	if !ok {
+		return "", fmt.Errorf("not canonical form: %q: %v", ref, err)
+	}
+
+	return canon.Digest().String(), nil
+}
+
+// compareOSImageURL determines whether two images are the same, or have
+// matching digests.
+func compareOSImageURL(current, desired string) (bool, error) {
```

Comment:
nit: The name of the function seems a little off but it's not a big deal. IE: this function is more so checking equality.

However: :+1: to the logic

Reply:
It's the same in the MCD.  Do you have a suggestion for a different name?  It's not quite about "equality" as at least in the MCD `""` was used to mean "unspecified" before although I stripped out that bit here.

Reply:
It's a nit, but something like `osImageHashMatch()` or something like that. It's not a big deal ... if the same name is in MCD then just leave as is.

Reply:
Yep, same as MCD.